hls4ml/model/optimizer/passes/stamp.py

```diff
@@ -0,0 +1,19 @@
+from hls4ml.model.optimizer import ModelOptimizerPass
+
+
+class MakeStamp(ModelOptimizerPass):
+    def __init__(self):
+        self.name = 'make_stamp'
+
+    def transform(self, model):
+        def _make_stamp():
+            """ Create a unique identifier for the generated code. This identifier is used to
+            compile a unique library and link it with python. """
+            from string import hexdigits
+            from random import choice
+            length = 8
+            return ''.join(choice(hexdigits) for m in range(length))
+
+        model.config.config['Stamp'] = _make_stamp()
```

Comment:
I know this isn't the point of this PR, but would it be safer to make the stamp a (hidden?) attribute of the model rather than a config entry?

Reply:
Yes! I was thinking that since layers have attributes, so should the model. Then all the config stuff that must be refactored can be implemented this way. But it's for the next PR.